package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -316,6 +316,18 @@ def __init__(self, universe,
 
         self.u = universe
         self._trajectory = self.u.trajectory
+
+        msg = ("{}_sel is an empty selection string - no hydrogen bonds will "
+               "be found. This may be intended, but please check your "
+               "selection."
+               )
+        for sel, sel_string in zip(
+            [donors_sel, hydrogens_sel, acceptors_sel],
+            ['donors', 'hydrogens', 'acceptors']
+        ):
+            if isinstance(sel, str) and not sel:
```

Comment:
Do you also want to test for the presence of blank space here? if so probably worth evaluating sel.strip()